akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MiscDirectivesSpec.scala

```diff
@@ -100,6 +100,23 @@ class MiscDirectivesSpec extends RoutingSpec {
       }
     }
 
+    "apply if form data is fully consumed into a map" in {
+      val route =
+        withSizeLimit(64) {
+          formFieldMap { _ ⇒
+            completeOk
+          }
+        }
+
+      Post("/abc", formDataOfSize(32)) ~> route ~> check {
+        status shouldEqual StatusCodes.OK
+      }
+
+      Post("/abc", formDataOfSize(128)) ~> Route.seal(route) ~> check {
+        status shouldEqual StatusCodes.BadRequest
```

Comment:
Can you add an assertion on the body as well? What kind of error message will the client see?

Reply:
Assert on the body has been added.